ouroboros-network/src/Ouroboros/Network/BlockFetch/Client.hs

```diff
@@ -290,6 +294,12 @@ blockFetchClient _version controlMessageSTM
             -- Add the block to the chain DB, notifying of any new chains.
             addFetchedBlock (castPoint (blockPoint header)) block
 
+            let slotNo_m = case pointSlot $ blockPoint header  of
+                               Origin -> Nothing
+                               (At s) -> Just s
+            forgeTime_m <- join <$> traverse (atomically . slotToTime) slotNo_m
```

Comment:
[`WithOrigin`](https://github.com/input-output-hk/cardano-base/blob/master/slotting/src/Cardano/Slotting/Slot.hs#L56) has a `Traversable` instance, there's no need to use `Maybe`.

Reply:
This is an outdated PR, `master` doesn't use `Maybe`. https://github.com/input-output-hk/ouroboros-network/blob/dea09273c87578a2047445e4434ba30df6570ac2/ouroboros-network/src/Ouroboros/Network/BlockFetch/Client.hs#L297